### ISSUE TITLE
Git version eden-pre

### ISSUE THREAD

**Issue by MartijnKaijser** [Giftie/script.cdartmanager#10]
When building database and you try to cancel it only cancels one part so it will continue with the next part.
To be more clear. If you cancel building album it will continue with building dtabase. if you cancel that it will skip to retrieving all artists.

When you cancel retrieving all artists it will say cancel but will just continue retrieving. So it's probably missing the xbmc.abortrequest

The update add-on database isn't available from the menu (it's probably on purpose because it's not ready?)

**Comment by Giftie** [comment]
It is missing the xbmcgui.DialogProcess().iscanceled() testing in the local artist area.  Will be added... :)

As you thought, it is not ready yet.  It is programmed but fails if you have duplicates of your albums, it will update both to the first path.  I need to figure out another way around.

**Comment by MartijnKaijser** [comment]
you mean when you have a normal edition and a special edition?

**Comment by Giftie** [comment]
Only if XBMC treats them as the same album(if you use MusicBrainz Picard then this happens easily).  This usually happens if you have the same album in two different formats(ie MP3 and FLAC)  

If XBMC had them as different albums, then all things are good.

**Comment by MartijnKaijser** [comment]
yes very difficult.
We have the same with movies. If you have let's say a bluray and a dvd version and the title are in genral the same so the add-on will only pick the first one. If i can remember it's only when the the skin provided the title to the add-on. so it will pick the first one that it finds in the database.

So far it's working rather well (at least it builds the database). I do remember when i first ran it i made some notes:
- when question asked delete local db and you say no you cannot do anything. Only escape works.
- no disable option in addon settings for using all artists so it's start building for every artist which can take ver very long.
- when finished creating database you have no control in gui. Hit ESCAPE and run the add-on again and it works like it should.

**Comment by MartijnKaijser** [comment]
Some things i noticed and latest version.
When first run and it starts building the database and you cancel it for some reason it keeps openening the add-on settings when you start it again.
When doing update db and you try to cancel it keeps running without ay way of interupting

**Comment by MartijnKaijser** [comment]
forgot to mention that on first db build and it finished the album part it went straight on to the all artists building without any questions asked (since i have 7000 artists that wasn't really fun so i did cancel)

**Comment by Giftie** [comment]
I thought I fixed that a while ago, but I see that I didn't.  I've added another setting that allows you to control the local artist part(it is defaulted to not add the local artists.)  Also I have added the ability to cancel. And I fixed the problem if you did cancel during the local artists part, that the database will be correctly closed.

**Comment by Giftie** [comment]
Just posted a new version - https://github.com/downloads/Giftie/script.cdartmanager/script.cdartmanager.2.5.5.zip

**Comment by MartijnKaijser** [comment]
perhaps a good  idea to integrate batch download i my screen keeps flickering with the downloaddialog on and off
or don't close the dialog and just do update (haven't really looked in the code yet so not sure)

**Comment by MartijnKaijser** [comment]
it still keeps opening the settings panel.  it finished downloading artist logos and after hitting OK it opened the settings panel again.
I know for sure the settings where already saved

**Comment by Giftie** [comment]
I think I have some what sorted out the problem with the Progress Dialog disappearing.  There were two problems: 
1. I have the script setting the percentage to 0, which causes the progress dialog to hide
2. I think might be an issue with the fanart.tv not reporting totalsize(ie Content-Length) for files being downloaded, since the progress bar does not move.  This was causing the percentage to be less than 0.  

I think I have fixed the redraw, so it shouldn't flash anymore.

I can't seem to reproduce the problem you are having with the settings opening up... Just to be sure I'm doing it correctly:

Your downloading the logo's using the automatic function( clearLOGOs -> Automatic Download -> Album Artists )  then it completes and presents a dialog showing the total number downloaded, then you click on the OK, then the settings open up?

**Comment by MartijnKaijser** [comment]
I never looked at file size with AD (which i am planning to do cause it looks nicer).

correct. It even opens when entering the  GUI of the add-on. you get quick progress dialog about retrieving the db and then the settings open.

here's the log if it would help  https://mail.google.com/mail/u/0/?ui=2&view=bsp&ver=ohhl4rw8mbn4

**Comment by Giftie** [comment]
I see why there is a problem..  you don't have a back up folder set.  The script tries to save a file 'missing.txt' to this folder and the script tests for this folder(opens setting if not found).  I have removed this requirement(now it will save just to the addon_data/script.cdartmanager folder.)

**Comment by MartijnKaijser** [comment]
why should i backup :P

**Comment by Giftie** [comment]
lol..  it's part of the old code when cdART Manager only handled cdARTs :)   

you are the best one for sourcing the bugs :) 

Thanks...

**Comment by MartijnKaijser** [comment]
You have no idea ;)

**Comment by MartijnKaijser** [comment]
any chance of pushing your latest changes to git so i can further bash them ;)

**Comment by Giftie** [comment]
Your wish is my command... :)

**Comment by MartijnKaijser** [comment]
hehe
You will hear from me if anything else comes up :P

**Comment by MartijnKaijser** [comment]
It takes a "long""  time if you hit download before the actual progress dialog appears. So i thought it didn't do anything and it cause it to run several things at a time.
Perhaps immidiate after you hit download show the dialog so you know it "worked".

**Comment by Giftie** [comment]
Yea.  When solving the problem for the progress dialog flashing, I put a test that first tries to update the dialog and if that fails, create the dialog.  The other problem is that there is no Content-Length, in the header of the download being supplied by fanart.tv(I have mentioned this to Kode) causing the download progress not to update.  

I moved the code up a few lines and this solved the problem.  Before I moved the code around, I was having XBMC hang after the download(using git)  After moving it the hang disappeared.  

new version :) - https://github.com/downloads/Giftie/script.cdartmanager/script.cdartmanager.2.5.7.zip
